src/Ontology/Transaction.h

```diff
@@ -0,0 +1,53 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include <string>
+
+#include "SigData.h"
+
+namespace TW {
+namespace Ontology {
+
+class Transaction {
+
+private:
+
+    uint8_t version;
+    uint8_t txType;
+    uint32_t nonce;
+    uint64_t gasPrice;
+    uint64_t gasLimit;
+    std::string payer;
+    std::vector<uint8_t> payload;
+    std::vector<uint8_t> attributes;
+
+public:
+
+    static const size_t sigVecLimit = 16;
+
+    std::vector<SigData> sigVec;
+
+    Transaction(uint8_t ver, uint8_t type, uint32_t nonce, uint64_t gasPrice, uint64_t gasLimit, std::string payer, std::vector<uint8_t> payload)
+            : version(ver), txType(type), nonce(nonce), gasPrice(gasPrice), gasLimit(gasLimit), payload(std::move(payload)) {
+        if (payer.empty()) {
+            payer = "AFmseVrdL9f9oyCzZefL9tG6UbvhPbdYzM";
```

Comment:
What's this address? could we make it as constant?

Reply:
This is based 58 encode zero address, we can make it as constant. What's name do you suggest?

Reply:
just zero address?

Reply:
right, zero address will be used to fill up payer data in pre-execute transaction. This type of transaction will not cost any ONG.